### PR TITLE
Adds Popover component (needed for D2IQ-65804)

### PR DESCRIPTION
**Pull request by mperrotti**
Relates to [D2IQ-65804](https://jira.d2iq.com/browse/D2IQ-65804)

## Testing
Take a look at the Popover stories under the "Overlays" heading

## Screenshots
Screen Shot 2020-04-20 at 6 16 24 PM

## Checklist for PR creator

- [x] If any new components were added, there are exported from `packages/index.ts`
- [ ] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
This might be intended, but I cannot copy the content in the menu.

**Comment by GeorgiSTodorov** [comment]
Also a minor thing, but the dimensions here are 2 pixels bigger than they should be.
Screenshot from 2020-04-21 10-16-46

File: packages/nonMenuDropdown/NonMenuDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+interface NonMenuDropdownProps
+  extends Omit<DropdownableProps, "open" | "dropdown" | "onClose"> {
+  /**
+   * A unique identifier to help screenreaders or UI testing tools associate the dropdown content with it's trigger button. If an ID prop is not provided, an arbitrary ID will be generated.
+   */
+  id?: string;
+  /**
+   * Whether the dropdown starts open
+   */
+  initialIsOpen?: boolean;
```
Comment (pierrebeitz) [inline packages/nonMenuDropdown/NonMenuDropdown.tsx:20]: how about `isInitiallyOpen`?

File: packages/nonMenuDropdown/NonMenuDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  /** The maximum height of the dropdown overlay, represented by number of pixels. */
+  maxHeight?: number;
```
Comment (pierrebeitz) [inline packages/nonMenuDropdown/NonMenuDropdown.tsx:22]: might there ever be the case that a `vh` could come in handy? in that case a string might also be a valid choice.

**Comment by danielone** [comment]
agree with @pierrebeitz  on the component naming
would recommend a positive naming vs. negative ;) 

Popover makes sense as different facets could launch it. 
https://material-ui.com/components/popover/
Check out how material design's also is related to modal
NonMenu naming feels confusing

**Comment by danielone** [comment]
"descriptive" aspect of the button naming in the Do's section feels too vague
Same button naming standards apply ( like verb + objectType, objectType, action depending on usage)
In this case, we will use [icon] Apps

**Comment by mperrotti** [comment]
Thank you all for the thoughtful feedback. I completely agree about the name, and have updated it to Popover.

All of the other feedback should be addressed too.

**Comment by GeorgiSTodorov** [comment]
> Also, here the content is hidden.

> Also a minor thing, but the dimensions here are 2 pixels bigger than they should be.

These two are still the same for me.

File: packages/nonMenuDropdown/NonMenuDropdown.tsx (hunk 1 above)
Comment (mperrotti) [reply packages/nonMenuDropdown/NonMenuDropdown.tsx]: I was trying to keep it consistent with `DropdownMenu`, which uses `initialIsOpen`

File: packages/nonMenuDropdown/NonMenuDropdown.tsx (hunk 2 above)
Comment (mperrotti) [reply packages/nonMenuDropdown/NonMenuDropdown.tsx]: There probably is a case, but I can't think of one right now. If we find a case for accepting strings with CSS units, we should do a follow-up PR that adds that functionality to all Popover-like components

**Pull request merged**